app/controllers/projects_controller.rb

```diff
@@ -135,18 +134,60 @@ def import_upload
     redirect_to [:import, @project]
   end
 
+  def reports
+    since = params[:since].nil? ? nil : params[:since].to_i.months.ago
+    @service = IterationService.new(@project, since)
+  end
+
+  def ownership
+    team = Team.not_archived.friendly.find(params.dig(:project, :slug))
+    if team == current_team
+      flash[:notice] = I18n.t('projects.invalid_action')
+      render "edit"
+      return
+    end
+
+    case params.dig(:ownership_action)
+    when 'share'
+      team.ownerships.create(project: @project, is_owner: false)
+      redirect_to edit_project_path(@project)
+    when 'unshare'
+      team.ownerships.where(project: @project).delete_all
+      redirect_to edit_project_path(@project)
+    when 'transfer'
+      Project.transaction do
+        current_team.ownerships.where(project: @project).delete_all
+        team.ownerships.create(project: @project, is_owner: true)
+      end
+      redirect_to root_path
+    else
+      flash[:alert] = I18n.t('projects.invalid_action')
+      redirect_to edit_project_path(@project)
+    end
+  end
+
   def archived
     @projects = policy_scope(Project).archived
     authorize @projects
   end
 
+  def archive
+    change_archived true
+  end
+
   def unarchive
+    change_archived false
+  end
+
+  protected
+
+  def change_archived(archive = true)
```

Comment:
since you don't call without param, the default value is useless, and based on the method's name,  if I see just `change_archived` I wouldn't know what value is assigning  to project achived